kikuchipy/signals/tests/test_ebsd_masterpattern.py

```diff
@@ -166,3 +180,136 @@ def test_set_phase_parameters(self):
         md_dict = s.metadata.get_item("Sample.Phases.1").as_dictionary()
         p.pop("number")
         assert_dictionary(p, md_dict)
+
+
+class TestEBSDCatalogue:
+
+    # Create  detector model
+    detector = EBSDDetector(
+        shape=(480, 640),
+        px_size=50,
+        pc=(20, 20, 15000),
+        convention="emsoft4",
+        tilt=10,
+        sample_tilt=70,
+    )
+
+    def test_get_direction_cosines(self):
+        out = _get_direction_cosines(self.detector)
+        assert isinstance(out, Vector3d)
+
+    def test_get_lambert_interpolation_parameters(self):
+        dc = _get_direction_cosines(self.detector)
+        scl = 500
+        npx = 1001
+        npy = 1001
+        (
+            nii,
+            nij,
+            niip,
+            nijp,
+            di,
+            dj,
+            dim,
+            djm,
+        ) = _get_lambert_interpolation_parameters(dc, scl, npx, npy)
+
+        assert (nii <= niip).all()
+        assert (nij <= nijp).all()
+
+        assert (nii < npx).all()
+        assert (nij < npy).all()
+        assert (niip < npx).all()
+        assert (nijp < npx).all()
+
+        assert (nii >= 0).all()
+        assert (nij >= 0).all()
+        assert (niip >= 0).all()
+        assert (nijp >= 0).all()
+
+    def test_get_patterns(self):
+
+        # Ni Test
+        EMSOFT_EBSD_FILE = os.path.join(
+            DIR_PATH, "../../data/emsoft_ebsd/EBSD_TEST_Ni.h5"
+        )
+
+        emsoft_key = load(EMSOFT_EBSD_FILE)
+
+        emsoft_key = emsoft_key.data[0]
+
+        angles = np.array((120, 45, 60))
+        r = Rotation.from_euler(np.radians(angles))
+
+        kp_mp = load(EMSOFT_MASTER, projection="lambert", hemisphere="both")
```

Comment:
This master pattern shouldn't be read like this, but instead

```python
kp_mp = kikuchipy.data.nickel_ebsd_master_pattern(
   projection="lambert", hemisphere="both"
)
```

Then you can remove the `EMSOFT_MASTER` string at the top as well.